internal/internal.go

```diff
@@ -106,6 +106,14 @@ var (
 	// This is used in the 1.0 release of gcp/observability, and thus must not be
 	// deleted or changed.
 	ClearGlobalDialOptions func()
+
+	// AddGlobalLateApplyDialOptions adds a slice of lateApplyDialOptions that
+	// will be configured for newly created ClientConns.
```

Comment:
This (or something) should explain what these magic things can or cannot do.

Reply:
I don't know what you mean by "can or cannot do". This is an arbitrary hack we added for one specific purpose, so I feel like scoping out what it's properties are and it's constraints doesn't apply? I added "This gets called after NewClient() parses the target, and allows per target
configuration set through a returned DialOption." to the PerTargetDialOption type (which I moved to internal/ to hide from users). Hopefully that's around what you meant by this comment?

Reply:
E.g. `WithResolvers` doesn't work.  Anything else?